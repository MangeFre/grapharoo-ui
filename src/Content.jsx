import React, { Component } from 'react';
import Header from './Header';

// Old way of making components
export default class Content extends Component {
	// Initializes the component
	constructor(props) {
		super(props);
		// initializing state - Infomartion we can change here.
		this.state = {
			history: [],
		};
		// Here I am also binding some methods that I want to be called from other
		// components. This is just to use this component as context so I can call this
		// in it.
		this.handleSubmit = this.handleSubmit.bind(this);
	}

	handleSubmit(submittedLink) {
		// Destructuring the state, to get an array of submitted links.
		console.log(`Submitted the link ${submittedLink} into Content component`);
		const { history } = this.state;
		history.push(submittedLink);
		// Cloning the array. There is some reason for this - I forgot why.
		const newHistory = [...history];
		this.setState({
			history: newHistory,
		});
	}

	// Renders the actual UI.
	render() {
		// Gonna grab the history from state and render it, if there is any.
		const { history } = this.state;
		return (
			<>
				<Header title="Grapharoo" subtitle="Mapping out the old Switcheroo Links" onSubmit={this.handleSubmit} />
				{history.map((oldLink) => {
					return <p key={oldLink}>{oldLink}</p>;
				})}
			</>
		);
	}
}
