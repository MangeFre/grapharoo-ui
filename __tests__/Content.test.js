import React from 'react';
import renderer from 'react-test-renderer';
import Content from '../src/Content';

test('Create snapshot for future comparison', () => {
    const component = renderer.create(<Content/>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})