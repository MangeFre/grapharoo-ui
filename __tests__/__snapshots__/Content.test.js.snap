// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create snapshot for future comparison 1`] = `
Array [
  <div
    className="jsx-250822500 header"
  >
    <div
      className="jsx-250822500 title"
    >
      <div
        className="jsx-250822500 icon"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-project-diagram fa-w-20 "
          data-icon="project-diagram"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M384 320H256c-17.67 0-32 14.33-32 32v128c0 17.67 14.33 32 32 32h128c17.67 0 32-14.33 32-32V352c0-17.67-14.33-32-32-32zM192 32c0-17.67-14.33-32-32-32H32C14.33 0 0 14.33 0 32v128c0 17.67 14.33 32 32 32h95.72l73.16 128.04C211.98 300.98 232.4 288 256 288h.28L192 175.51V128h224V64H192V32zM608 0H480c-17.67 0-32 14.33-32 32v128c0 17.67 14.33 32 32 32h128c17.67 0 32-14.33 32-32V32c0-17.67-14.33-32-32-32z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <div
        className="jsx-250822500"
      >
        <h1
          className="jsx-250822500"
        >
          Grapharoo
        </h1>
        <h3
          className="jsx-250822500"
        >
          Mapping out the old Switcheroo Links
        </h3>
      </div>
    </div>
    <div
      className="jsx-250822500 searchForm"
    >
      <form
        className="jsx-457726797 searchForm"
        name="link"
        onSubmit={[Function]}
      >
        <input
          className="jsx-457726797"
          name="linkInput"
          placeholder="Switcheroo Link"
          type="text"
        />
        <button
          className="jsx-457726797"
          type="submit"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-search fa-w-16 "
            data-icon="search"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </form>
    </div>
  </div>,
  <div
    className="jsx-1028534497"
  >
    <h1
      className="jsx-1028534497"
    >
      Enter a link up top
    </h1>
  </div>,
]
`;
